pxtblocks/code-validation/rubricCriteria.ts

```diff
@@ -0,0 +1,56 @@
+namespace pxt.blocks {
+    interface BlockSet {
+        blocks: string[];
+        count: number;
+    }
+
+    export interface CriteriaData {
+        criteriaID: string;
+        friendlyText: string;
+        blockRequirements?: BlockSet[];
+        count?: number;
+    }
+
+    export function getCriteria(data: CriteriaData): RubricCriteria {
+        switch (data.criteriaID) {
+            case "blockCheck":
+                return new BlockCheckCriteria(data.friendlyText, data.blockRequirements);
+            case "comment":
+                return new CommentCriteria(data.friendlyText, data.count);
+            default:
+                console.error(`Unrecognized criteriaID: ${data.criteriaID}`);
+                return null;
+        }
+    }
+
+    export abstract class RubricCriteria {
+        friendlyText: string;
+        abstract criteriaID: string;
```

Comment:
the smallest of nits: use `Id`, not `ID`:

```suggestion
        abstract criteriaId: string;
```